python/cudf/cudf/core/series.py

```diff
@@ -4106,26 +4106,31 @@ def isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
     dtype: bool
     """
 
+    if isinstance(a, (cudf.Series, pd.Series)) and isinstance(
+        b, (cudf.Series, pd.Series)
+    ):
+        b = b.reindex(a.index)
+
     a_col = column.as_column(a)
-    a_array = a_col.to_array(fillna=-1 if a_col.null_count else None)
+    a_array = cupy.array(a_col.data_array_view)
```

Comment:
`cupy.asarray` gives zero copy semantics:
```suggestion
    a_array = cupy.asarray(a_col.data_array_view)
```